File: Leetcode75/345_Reverse_vowels.java
class Solution {
    public String reverseVowels(String s) {
        char[] word = s.toCharArray();
        String vowels = "AaEeIiOoUu";
        int start = 0;
        int end = word.length - 1;
        while (start < end) {
            while ((start < end) && (vowels.indexOf(word[start]) == -1)) {
                start++;
            }
            while ((start < end) && (vowels.indexOf(word[end]) == -1)) {
                end--;
            }
            if (start < end) {
                char temp = word[start];
                word[start] = word[end];
                word[end] = temp;
            }
            start++;
            end--;
        }
        return new String(word);
    }
}

File: Leetcode75/345_Reverse_Vowels.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = "AaEeIiOoUu"
        vowelFound = []
        vowelIndexes = []
        for i, char in enumerate(s):
            if char in vowels:
                vowelIndexes.append(i)
                vowelFound.append(char)
        
        s = list(s)
        for i in vowelIndexes:
            s[i] = vowelFound.pop(-1)
        return "".join(s)
        
